### PR TITLE
Ejinks

### PR DESCRIPTION
**Pull request by ejinks**
Creating this pull request to implement liking functionality for both libraries and comments, ajax for posting comments, and functionality for editing and deleting comments

### CODE REVIEW

File: routes/libraries.js [hunk 1]
```diff
@@ -175,7 +175,8 @@ router.route('/:id')
     }
 
     try {
-      res.render('libraries/library', { title: library.name, library: library, isLoggedIn: true, script_partial: 'comment', userid: req.session.user._id});
+      let user = req.session.user;
+      res.render('libraries/library', { title: library.name, isLoggedIn: true, script_partial: 'comment', userid: user._id, ...library});
```
Comment (galapatt) [inline routes/libraries.js]: user should be saved as a global variable so you can access directly through handlebars without passing it thru

File: routes/libraries.js [hunk 2]
```diff
@@ -297,9 +298,9 @@ router.route('/:id/comments')
     }
 
     try {
-      let userid = req.session.user._id.toString();
-      let createComment = await libraryData.createComment(id, userid, text);
-      res.redirect(`libraries/library/${id}`);
+      let user = req.session.user;
+      let createComment = await libraryData.createComment(id, user._id, user.userName, text);
+      res.render('partials/comment', {layout: null, library, ...createComment});
```
Comment (galapatt) [inline routes/libraries.js]: same for this, but works regardless

**Review by galapatt — COMMENTED** [review]
all goods, just offered some suggestions

**Comment by JulienCarr2** [comment]
Looks fine, just make sure to remember to handle the conflicts.

File: routes/libraries.js (hunk 1 above)
Comment (ejinks) [reply routes/libraries.js]: What do you mean?

**Comment by DanielleDauphinais** [comment]
Looks good but there seems to be a merge conflict so when we are merging have to work on that. One little thing is that when ever we do res.render we need to have a title defined but I will go back and do that to everything when we are "cleaning" the code.

File: public/js/comments.js [hunk 3]
```diff
@@ -3,49 +3,44 @@ import {validationsForStrings} from './validators/util.js';
...
+
     try {
-      text = text.val();
+      text = textArea.val();
       validationsForStrings("Comment body", text);
     } catch (e) {
+      console.log(e);
+      console.log(error);
       error.hidden = false;
       error.innerHTML = e;
```
Comment (ejinks) [inline public/js/comments.js]: Anyone know why this isn't rendering on the screen?

File: data/users.js [hunk 4]
```diff
@@ -42,6 +43,7 @@ let exportedMethods = {
     // Add libraryId to user's favorited libraries
     const userCollection = await users();
 
+    const user = await userCollection.findOne({ _id: new ObjectId(userId) });
     const user = await userCollection.findOne({ _id: new ObjectId(userId) });
```
Comment (vishwa5854) [inline data/users.js]: Hey the server is crashing because of this since u are redeclaring a const
please check this

**Review by vishwa5854 — CHANGES_REQUESTED** [review]

**Pull request merged**